google-cloud-storage/lib/google/cloud/storage/bucket.rb

```diff
@@ -3143,6 +3143,50 @@ def reload!
         end
         alias refresh! reload!
 
+        ##
+        # Moves File from source to destination path within bucket
```

Comment:
We should also mention that this works only for HNS buckets. You can take reference from https://github.com/googleapis/nodejs-storage/pull/2586/files#diff-bc9705d0f7a567399044dfc66ccc82d4d9aa1cff116842a0094d54e463c9ecbcR3474